### PR TITLE
[FLINK-14560] [runtime] The value of taskmanager.memory.size in flink…

### PR DESCRIPTION
**Pull request by faaronzheng**
…-conf.yaml is set to zero will cause taskmanager not to work

## What is the purpose of the change

In order to fix the issue FLINK-14560

## Brief change log
TaskManagerOptions.java ：remove the default value
ResourceManager.java: modify the code to suit none value
ConfigurationParserUtils.java: modify the code  to suit none value
TaskManagerHeapSizeCalculationJavaBashTest.java: modify the code  to suit none value 
NetworkBufferCalculationTest.java: modify the code  to suit none value 
LocalStreamEnvironment.java: remove TaskManagerOptions.MANAGED_MEMORY_SIZE with value 0
SavepointITCase.java: remove TaskManagerOptions.MANAGED_MEMORY_SIZE with value 0

## Verifying this change

This change is already covered by existing tests, such as (please describe tests).
TaskManagerHeapSizeCalculationJavaBashTest
NettyShuffleEnvironmentConfigurationTest
NetworkBufferCalculationTest
YarnConfigurationITCase
SavepointITCase

## Does this pull request potentially affect one of the following parts:

Dependencies (does it add or upgrade a dependency): (no)
The public API, i.e., is any changed class annotated with @public(Evolving): (no)
The serializers: (no)
The runtime per-record code paths (performance sensitive): ( no)
Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (yes)
The S3 file system connector: (no)

## Documentation

Does this pull request introduce a new feature? (no)
If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

**Comment by faaronzheng** [comment]
> * @flinkbot run travis

@flinkbot run travis

**Comment by faaronzheng** [comment]
> * @flinkbot run travis

@flinkbot run travis

**Comment by faaronzheng** [comment]
> ## CI report:
> * [f7802ae](https://github.com/apache/flink/commit/f7802aefa43e285597bfba62592e05d0cb2f074f) : FAILURE [Build](https://travis-ci.com/flink-ci/flink/builds/137937387)
> 
> Bot commands

@flinkbot run travis

**Comment by faaronzheng** [comment]
> * @flinkbot run travis

@flinkbot run travis

**Comment by faaronzheng** [comment]
>     * @flinkbot run travis

@flinkbot run travis

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java [hunk 1]
```diff
@@ -42,25 +42,20 @@
...
-			"MemoryManager needs at least one MB of memory. " +
-				"If you leave this config parameter empty, the system automatically pick a fraction of the available memory.");
 
+			checkConfigParameter( managedMemorySize > 0,
+				managedMemorySize, TaskManagerOptions.MANAGED_MEMORY_SIZE.key(),
+				"MemoryManager needs at least one MB of memory. " +
+					"If you leave this config parameter empty, the system automatically pick a fraction of the available memory.");
+		}
 		return managedMemorySize;
 	}
```
Comment (tillrohrmann) [inline flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java]: Would this work instead?

```
public static long getManagedMemorySize(Configuration configuration) {
		final MemorySize defaultManagedMemory = MemorySize.parse(TaskManagerOptions.MANAGED_MEMORY_SIZE.defaultValue());

		final String managedMemoryValue = configuration.getString(TaskManagerOptions.MANAGED_MEMORY_SIZE);
		final MemorySize configuredManagedMemory;

		try {
			configuredManagedMemory = MemorySize.parse(managedMemoryValue, MEGA_BYTES);
		} catch (IllegalArgumentException e) {
			throw new IllegalConfigurationException("Could not read " + TaskManagerOptions.MANAGED_MEMORY_SIZE.key(), e);
		}

		checkConfigParameter(defaultManagedMemory.equals(configuredManagedMemory) || configuredManagedMemory.getMebiBytes() > 0,
			managedMemoryValue, TaskManagerOptions.MANAGED_MEMORY_SIZE.key(),
			"MemoryManager needs at least one MB of memory. " +
				"If you leave this config parameter empty, the system automatically pick a fraction of the available memory.");

		return configuredManagedMemory.getMebiBytes();
	}
```

**Comment by tillrohrmann** [comment]
I've pushed an update of the PR with updated changes.

File: flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java (hunk 1 above)
Comment (faaronzheng) [reply flink-runtime/src/main/java/org/apache/flink/runtime/util/ConfigurationParserUtils.java]: sure

**Pull request closed without merging**